addons/project/models/project.py

```diff
@@ -2223,3 +2224,19 @@ def _get_default_color(self):
     _sql_constraints = [
         ('name_uniq', 'unique (name)', "Tag name already exists!"),
     ]
+
+    @api.model
+    def _name_search(self, name='', args=None, operator='ilike', limit=100, name_get_uid=None):
+        domain = [('name', operator, name)]
+        if 'project_id' in self.env.context.keys():
+            project_id = self.env.context.get('project_id')
+            tag_ids = self.env['project.task'].search([('project_id', '=', project_id)]).mapped('tag_ids').ids + self.env['project.project'].search([('id', '=', project_id)]).mapped('tag_ids').ids
```

Comment:
Instead of doing this we could define two fields, one using the same table defined in the m2m field containing the `tag_ids` in the project model and the second using the table defined in the m2m fields containing the `tag_ids` linked to the task model.

When we have defined those fields we could search the task_ids of the project found in the context and then all those fields in the domain instead of `('id', 'in', tag_ids)`. 🙂

I hope it is clear for you. 😇

Reply:
I'm not sure to understand what would be the advantage in doing this instead? What would we gain?

Reply:
We could have the projects linked to the tag_ids and also the tasks with those fields in fact. ^^
And thus, we could avoid to do this search: `self.env['project.project'].search([('id', '=', project_id)]).mapped('tag_ids').ids`. 😇

Reply:
It should be better now, but let me know what you think